core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -387,6 +388,14 @@ private[spark] class MemoryStore(
 
   def remove(blockId: BlockId): Boolean = memoryManager.synchronized {
     val entry = entries.synchronized {
+      entries.get(blockId) match {
+        case e: DeserializedMemoryEntry[_] => e.value.foreach(v => v match {
+          case o: AutoCloseable =>
+              o.close
+            case other =>
```

Comment:
Nit: indentation, and just write _, not other
Also put parens on .close; it has side effects.